city/crud.py

```diff
@@ -0,0 +1,59 @@
+from sqlalchemy.future import select
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from city import models, schemas
+
+
+async def post_city(db: AsyncSession, city_schema: schemas.CityCreate):
+    city = models.City(
+        name=city_schema.name,
+        additional_info=city_schema.additional_info
+    )
+    db.add(city)
+    await db.commit()
+    await db.refresh(city)
+    return city
+
+
+async def get_all_cities(db: AsyncSession) -> list[models.City]:
+    city_list = await db.execute(select(models.City))
+    return [city[0] for city in city_list.fetchall()]
+
+
+async def get_city_detail(db: AsyncSession, city_id: int) -> models.City | None:
```

Comment:
The return type hint `models.City | None` uses the new union operator introduced in Python 3.10. If you need to support older Python versions, consider using `Optional[models.City]` from the `typing` module.